Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Operator/FleetManagement.hs

```diff
@@ -135,50 +146,110 @@ castFleetType = \case
   Common.NORMAL_FLEET -> FOI.NORMAL_FLEET
   Common.BUSINESS_FLEET -> FOI.BUSINESS_FLEET
 
-postFleetManagementFleetLink ::
+checkOperator :: Text -> Environment.Flow DP.Person
+checkOperator requestorId = do
+  operator <- B.runInReplica $ QP.findById (Id requestorId :: Id DP.Person) >>= fromMaybeM (PersonNotFound requestorId)
+  unless (operator.role == DP.OPERATOR) $
+    throwError AccessDenied
+  pure operator
+
+postFleetManagementFleetUnlink ::
   Kernel.Types.Id.ShortId Domain.Types.Merchant.Merchant ->
   Kernel.Types.Beckn.Context.City ->
   Kernel.Prelude.Text ->
   Kernel.Prelude.Text ->
   Environment.Flow Kernel.Types.APISuccess.APISuccess
-postFleetManagementFleetLink merchantShortId opCity fleetOwnerId requestorId = do
-  operator <- B.runInReplica $ QP.findById (Id requestorId :: Id DP.Person) >>= fromMaybeM (PersonNotFound requestorId)
-  unless (operator.role == DP.OPERATOR) $
-    throwError AccessDenied
+postFleetManagementFleetUnlink _merchantShortId _opCity fleetOwnerId requestorId = do
+  operator <- checkOperator requestorId
 
   fleetOwner <- B.runInReplica $ QP.findById (Id fleetOwnerId :: Id DP.Person) >>= fromMaybeM (FleetOwnerNotFound fleetOwnerId)
   unless (fleetOwner.role == DP.FLEET_OWNER) $
     throwError (InvalidRequest "Invalid fleet owner")
 
+  QFOA.endFleetOperatorAssociation fleetOwner.id operator.id
+
+  -- TODO send sms about unlink
+  pure Kernel.Types.APISuccess.Success
+
+postFleetManagementFleetLinkSendOtp ::
+  Kernel.Types.Id.ShortId Domain.Types.Merchant.Merchant ->
+  Kernel.Types.Beckn.Context.City ->
+  Kernel.Prelude.Text ->
+  Common.FleetOwnerSendOtpReq ->
+  Environment.Flow Common.FleetOwnerSendOtpRes
+postFleetManagementFleetLinkSendOtp merchantShortId opCity requestorId req = do
+  operator <- checkOperator requestorId
+  merchant <- findMerchantByShortId merchantShortId
+  mobileNumberHash <- getDbHash req.mobileNumber
+  fleetOwner <-
+    B.runInReplica $
+      QP.findByMobileNumberAndMerchantAndRole req.mobileCountryCode mobileNumberHash merchant.id DP.FLEET_OWNER
+        >>= fromMaybeM (FleetOwnerNotFound "Fleet owner with this phone number not found")
+
   checkAssocOperator <- B.runInReplica $ QFOA.findByFleetOwnerIdAndOperatorId fleetOwner.id operator.id True
   when (isJust checkAssocOperator) $ throwError (InvalidRequest "Fleet already associated with operator")
 
-  merchant <- findMerchantByShortId merchantShortId
+  smsCfg <- asks (.smsCfg)
   merchantOpCityId <- CQMOC.getMerchantOpCityId Nothing merchant (Just opCity)
-  SA.endFleetAssociationsIfAllowed merchant merchantOpCityId fleetOwner
```

Comment:
lets add this check here, so we can overwrite association if it is allowed 

`SA.endFleetAssociationsIfAllowed merchant merchantOpCityId fleetOwner`